lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1257,10 +1257,8 @@ static void replaceOperandWithAnnotationIntrinsicCallResult(Function *F,
     for (auto *Use : BV->users()) {
       if (auto *II = dyn_cast<IntrinsicInst>(Use)) {
         if (II->getIntrinsicID() == Intrinsic::ptr_annotation &&
-            II->getType() == BV->getType()) {
```

Comment:
The SPEC says: It is invalid to decorate any given <id> or structure member more than one time with the same [decoration](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#Decoration), unless explicitly allowed below for a specific decoration
However, there is no restriction about decorating any given <id> with two different decorations. So, I think we can remove this assert.
Please provide feedback if the SPEC has not been interpreted correctly.
Thanks

Reply:
@broxigarchen @asudarsa is this thread resolved?

Reply:
Yes. We agreed that this is something we may address in the future. But it does not block the current PR.

Thanks